src/module/mod.rs

```diff
@@ -227,13 +227,13 @@ impl Module {
     where
         P: AsRef<Path>,
     {
-        let buffer = self.emit_wasm()?;
+        let (buffer, _) = self.emit_wasm(true)?;
         fs::write(path, buffer).context("failed to write wasm module")?;
         Ok(())
     }
 
     /// Emit this module into an in-memory wasm buffer.
-    pub fn emit_wasm(&self) -> Result<Vec<u8>> {
+    pub fn emit_wasm(&self, with_map: bool) -> Result<(Vec<u8>, Option<Vec<(usize, usize)>>)> {
```

Comment:
I'd feel a little better about doing the configuration in the `ModuleConfig` (`src/module/config.rs`) and then having two methods here:

1. get the in-memory wasm buffer alone
2. get the in-memory wasm buffer and the offsets map together (maybe even panics if it wasn't configured to track that info)

This avoids adding semi-magical boolean parameters that most people don't want to think about to functions that most people need to call.

Reply:
Added `ModuleConfig.preserve_code_transform`